src/accel/accelCk.ts

```diff
@@ -0,0 +1,96 @@
+const AccelMsg_ck = `
+public class AccelMsg {
+    float accelX;
+    float accelY;
+    float accelZ;
+
+    function float getAccelX() {
+        return accelX;
+    }
+
+    function float getAccelY() {
+        return accelY;
+    }
+
+    function float getAccelZ() {
+        return accelZ;
+    }
+
+    function void _copy(AccelMsg localMsg) {
+        localMsg.accelX => accelX;
+        localMsg.accelY => accelY;
+        localMsg.accelZ => accelZ;
+    }
+}
+`;
+
+const Accel_ck = `
+global Event _accelReading;
+global int _accelActive;
+
+global float _accelX;
+global float _accelY;
+global float _accelZ;
+
+public class Accel extends Event {
+
+    0 => int isAccelOpen;
+    0 => int active;
+
+    string deviceName; 
+
+    // AccelMsg Queue
+    AccelMsg _accelMsgQueue[0];
+
+    function string name() {
+        return deviceName;
+    }
+
+    function int openAccel(int num) {
+        if (num < 0) {
+            false => active;
+        } else {
+            "js DeviceMotionEvent" => deviceName;
```

Comment:
nit: Is the device name `js DeviceMotionEvent` or more like `js DeviceMotionSensor`?

Reply:
I don't think there is a specific device name... it's just representing access to device orientation and acceleration. I made up this shorthand reference... we could just call it "accelerometer" or "gyroscope" as that is essentially what we are accessing. What do you think?